control-plane-operator/controllers/awsprivatelink/route53.go

```diff
@@ -83,48 +78,39 @@ func fqdn(name string) string {
 	}
 }
 
-func findRecord(ctx context.Context, client route53iface.Route53API, id, name string) (*route53.ResourceRecordSet, error) {
+func findRecord(ctx context.Context, client *route53.Client, id, name string) (*route53types.ResourceRecordSet, error) {
 	recordName := fqdn(strings.ToLower(name))
-	recordType := "CNAME"
 	input := &route53.ListResourceRecordSetsInput{
 		HostedZoneId:    aws.String(id),
 		StartRecordName: aws.String(recordName),
-		StartRecordType: aws.String(recordType),
-		MaxItems:        aws.String("1"),
+		StartRecordType: route53types.RRTypeCname,
+		MaxItems:        aws.Int32(1),
+	}
+	resp, err := client.ListResourceRecordSets(ctx, input)
```

Comment:
I started looking into this because the new client doesn't have a paginator for record sets: https://github.com/openshift/hypershift/pull/892

I noticed that we filter by name and type and have MaxItems: 1, what was the reason to use pagination in the first place? Is it possible to get more than one page with MaxItems: 1?